packages/sdk/src/flags.ts

```diff
@@ -0,0 +1,46 @@
+import { Contextual } from '.';
+import { Value } from './Value';
+
+export namespace FlagEvaluation {
+  export interface Matched<T> {
+    readonly reason: 'MATCH';
+    readonly value: T;
+    readonly variant: string;
+  }
+
+  export interface Unmatched<T> {
+    readonly reason:
+      | 'UNSPECIFIED'
+      | 'NO_SEGMENT_MATCH'
+      | 'NO_TREATMENT_MATCH'
+      | 'FLAG_ARCHIVED'
+      | 'TARGETING_KEY_ERROR';
+    readonly value: T;
+  }
+
+  export interface Failed<T> {
+    readonly reason: 'ERROR';
+    readonly value: T;
+    // TODO Change PROVIDER_NOT_READY to NOT_READY
```

Comment:
we should probably address this now (or actually maybe in a separate PR before crafting release)